### PR TITLE
Test makefile

### PR DESCRIPTION
**Pull request by Juan-M-V**
# Makefile test rule

## Description

Add setup to tests

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -14,7 +14,9 @@ check:
 	cargo check
 
 test:
+	cd tests; ./setup_tests
 	cargo test
+	rm cairo_programs/*.json
```
Comment (pefontana) [inline Makefile:19]: I think it would be more clear if  we add this removes in the setup_tests script

File: Makefile [hunk 2]
```diff
@@ -14,7 +14,9 @@ check:
 	cargo check
 
 test:
+	cd tests; ./setup_tests
```
Comment (pefontana) [inline Makefile:17]: lets rename this `setup_tests` to `setup_tests.sh`

File: Makefile (hunk 1 above)
[the remark this answers is not in the document]
Comment (Oppen) [reply Makefile:19]: `Make` simply invokes a shell to run the commands (after expression evaluation) so whatever changes of directories don't propagate outside of the target commands.

To illustrate, here's a `Makefile`:
```
test_call: test
	pwd

test:
	cd s5cmd; pwd

.PHONY: test test_call
```

If you run `make test_call`, which depends on `test` and thus triggers its build, you'll see output like this (mine is at the root of my code directory):
```
cd s5cmd; pwd
/Users/mario.rugiero/Code/s5cmd
pwd
/Users/mario.rugiero/Code
```

Note I never use `cd ..` and yet building the dependent target happens in the original dir.

**Comment by pefontana** [comment]
@Juan-M-V one last thing :P
Please add a line to `.gitignore` that prevents pushing the generated cairo_programs/*.json

**Comment by unbalancedparentheses** [comment]
> @Juan-M-V one last thing :P
> 
> Please add a regex match in the `.gitignore` that prevents pushing the generated `cairo_programs/*.json`

great comment. please create new pr for this

**Pull request merged**